R/utils.R

```diff
@@ -330,6 +350,249 @@ markdown_to_latex <- function(text) {
     unname()
 }
 
+cmark_rules <- list(
+
+  heading = function(x, process) {
+    heading_sizes <- c(36, 32, 28, 24, 20, 16)
+    fs <- heading_sizes[as.numeric(xml2::xml_attr(x, attr = "level"))]
+
+    rtf_paste0(
+      rtf_raw("{\\ql \\f0 \\sa180 \\b \\fs"),
+      rtf_raw(fs),
+      rtf_raw(" "),
+      process(xml2::xml_children(x)),
+      rtf_raw("}")
+    )
+  },
+  thematic_break = function(x, process) {
+    rtf_raw("{\\qc \\f0 \\sa180 \\emdash\\emdash\\emdash\\emdash\\emdash}")
+  },
+  link = function(x, process) {
+    # NOTE: RTF doesn't handle the `title` attr (Pandoc also ignores)
+    destination <- xml2::xml_attr(x, attr = "destination")
+    text <- process(xml2::xml_children(x))
+    if (text == "") text <- destination
+    rtf_raw("{\\field{\\*\\fldinst{HYPERLINK \"", destination, "\"}}{\\fldrslt{\\ul ", text, "}}}")
+  },
+  list = function(x, process) {
+
+    type <- xml2::xml_attr(x, attr = "type")
+    n_items <- length(xml2::xml_children(x))
+
+    # NOTE: `start`, `delim`, and `tight` attrs are ignored; we also
+    # assume there is only `type` values of "ordered" and "bullet" (unordered)
+    rtf_raw(
+      paste(
+        "\\pard\\intbl\\itap1\\tx220\\tx720\\tx1133\\tx1700\\tx2267\\tx2834\\tx3401\\tx3968\\tx4535\\tx5102\\tx5669\\tx6236\\tx6803\\li720\\fi-720",
```

Comment:
Interesting that "intbl" is necessary (I'm assuming) here. But we don't need it for other paragraph rules?

Reply:
Sorry, I see now that blockquote has it too.